modules/common/src/main/scala/notebook/front/widgets/magic/Magic.scala

```diff
@@ -27,42 +26,50 @@ trait MagicRenderPoint { me =>
-      case v: Long        => Seq(header.getOrElse("Long"))
-      case v: BigDecimal  => Seq(header.getOrElse("BigDecimal"))
-      case v: String      => Seq(header.getOrElse("String"))
-      case v: Boolean     => Seq(header.getOrElse("Boolean"))
-      case v: SqlDate     => Seq(header.getOrElse("Date"))
-      case v: SqlTimestamp => Seq(header.getOrElse("Timestamp"))
-      case v: Date        => Seq(header.getOrElse("Date"))
-      case v: Geometry    => Seq(header.getOrElse("Geometry"))
-      case v: GeoJSON     => Seq(header.getOrElse("GeoJSON"))
-      case v: Any         => Reflector.toFieldNameArray(any)
+      case _: Int         => Seq(header.getOrElse("Int"))
+      case _: Float       => Seq(header.getOrElse("Float"))
+      case _: Double      => Seq(header.getOrElse("Double"))
+      case _: Long        => Seq(header.getOrElse("Long"))
+      case _: BigDecimal  => Seq(header.getOrElse("BigDecimal"))
+      case _: String      => Seq(header.getOrElse("String"))
+      case _: Boolean     => Seq(header.getOrElse("Boolean"))
+      case _: SqlDate     => Seq(header.getOrElse("Date"))
+      case _: SqlTimestamp => Seq(header.getOrElse("Timestamp"))
+      case _: Date        => Seq(header.getOrElse("Date"))
+      case _: Geometry    => Seq(header.getOrElse("Geometry"))
+      case _: GeoJSON     => Seq(header.getOrElse("GeoJSON"))
+      case _: Any         => Reflector.toFieldNameArray(point)
   }
-  val values  = any match {
+
+  /**
+    * Result of this conversion is displayed in widgets (like in [[notebook.front.widgets.charts.TableChart]] for
+    * example. Thus we want to convert points containing dates into human readable format. Otherwise they would be
+    * displayed as timestamps.
+    *
+    * This is also consumed by c3 library when plotting [[notebook.front.widgets.charts.TimeseriesChart]]. So string
+    * format used here needs to be in sync with format used to parse this string there.
+    */
+  override val values  = point match {
       case null           => Seq("<null-value>")
       case v: Int         => Seq(v)
       case v: Float       => Seq(v)
       case v: Double      => Seq(v)
       case v: Long        => Seq(v)
       case v: BigDecimal  => Seq(v)
-      case v: SqlDate     => Seq(v.toString)
-      case v: SqlTimestamp => Seq(v.toString)
-      case v: Date        => Seq(v.toString)
+      case v: Date        => Seq(dateFormat.format(v))
       case v: String      => Seq(v)
       case v: Boolean     => Seq(v)
       case v: Geometry    => Seq(v)
       case v: GeoJSON     => Seq(v)
-      case v: Any         => Reflector.toFieldValueArray(any).map {
-        // convert fields into "human-readable" format
-        case fieldValue @ (_: SqlDate | _: SqlTimestamp | _: Date) =>
-          // otherwise these would be displayed as integer timestamps...
-          fieldValue.toString
+      case v: Any         => Reflector.toFieldValueArray(v).map {
+        case v: Date => dateFormat.format(v)
```

Comment:
maybe it would be better to explicitly format `SqlDate/SqlTimestamp` here too (instead of relying on their default formats)?

I guess this was referring to `case class` fields (`TableChart(Seq(someCaseClass))`).

Reply:
This is not using default types. Both `java.sql.Date` and `java.sql.Timestamp` are subclasses of `java.util.Date`. This one case on line 72 is checking for type `java.util.Date` and will be used for those two sql types. Maybe that's what you meant in question above...